### PR TITLE
#968 Add VMSS to run web servers and Add Azure Files share

### PR DESCRIPTION
**Pull request by lawrencegripper**
/w AutoUpdatingOS, AutoScaling, Rolling update, Loadbalancer with Healthchecks, cloudinit docker install

### CODE REVIEW

**Comment by lawrencegripper** [comment]
I've got this together an running to do some basic tests

- [ ] Healthprobe detects servers not running and responding on '/webservices/rest/v1/meta/status` @wbari is that the right endpoint?
- [ ] Update uses rolling so waits for healthy nodes not taking out all nodes at the same time
- [ ] Integrated with Avere through main template and script to mount the storage at `/media/primarynfs`

**Comment by lawrencegripper** [comment]
This is now working with Azure files too, there is outstanding work in #965 to fix the usability of Avere once we are able to successfully deploy it.

File: opencga-app/app/scripts/azure/arm/azuredeploy.json [hunk 1]
```diff
@@ -85,6 +91,24 @@
...
+        "webserversSSHKeyData": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The RSA data for the SSH Public Key used to authenticate to Avere nodes."
+            }
+        },
+        "tomcatDockerImage" : {
+            "type": "string",
+            "metadata": {
+                "description": "The docker image and tag for running the tomcat service."
```
Comment (jjcollinge) [inline opencga-app/app/scripts/azure/arm/azuredeploy.json]: Appreciate the description is pretty clear but would be good to have an example string that meets the expected format on these i.e. repo/image:tag

File: opencga-app/app/scripts/azure/arm/azuredeploy.json [hunk 2]
```diff
@@ -244,12 +285,51 @@
...
+                        "value": "[parameters('_artifactsLocation')]"
+                    },
+                    "_artifactsLocationSasToken": {
+                        "value": "[parameters('_artifactsLocationSasToken')]"
+                    },
+                    "virtualNetworkSubnetId": {
+                        "value": "[reference('vnet').outputs.avereClusterSubnetId.value]"
+                    },
+                    "mountArgs": {
+                        "value": "[concat('azurefiles ', reference('azurefiles').outputs.accountName.value, ',', reference('azurefiles').outputs.shareName.value, ',' , reference('azurefiles').outputs.accountKey.value)]"
```
[the remark this answers is not in the document]
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/azuredeploy.json]: Yes the python expects sudo `python3 mount.py azurefiles <storage-account-name>,<share-name>,<storage-account-key>` for azurefiles mounting

File: opencga-app/app/scripts/azure/arm/avere/azuredeploy.json [hunk 3]
```diff
@@ -172,6 +172,7 @@
       }
     },
     {
+      "comments": "This creates a controller VM which is responsible for provisioning the Avere cluster nodes. The binary provided in the cloud init is the boostrap script from here: https://github.com/Azure/Avere/tree/a2e28b5cbc2ca467d10acbf12abfd6f8bf5edcba",
```
[the remark this answers is not in the document]
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/avere/azuredeploy.json]: :smile:

File: opencga-app/app/scripts/azure/arm/azuredeploy.json [hunk 4]
```diff
@@ -62,6 +62,12 @@
                 "description": "The password must be at least 10 characters in length and must contain at least one digit, one non-alphanumeric character, and one upper or lower case letter."
             }
         },
+        "avereEnabled": {
```
[the remark this answers is not in the document]
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/azuredeploy.json]: Yes, good call. I'll add this now.

File: opencga-app/app/scripts/azure/arm/azuredeploy.json (hunk 1 above)
Comment (lawrencegripper) [reply opencga-app/app/scripts/azure/arm/azuredeploy.json]: Deal, will add it

File: opencga-app/app/scripts/azure/arm/azuredeploy.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjcollinge) [reply opencga-app/app/scripts/azure/arm/azuredeploy.json]: Ok, I'll make sure the daemonvm executes the script in a similar manner then to save rehashing the args

**Comment by lawrencegripper** [comment]
Just a quick note, when testing this I've been using `ricc/http-devnull` as the image for both IVA and Tomcat as it responds to any request on any path with `200` 

Without it the VMSS nodes show as unhealthy and this prevents any rolling updates.

**Comment by lawrencegripper** [comment]
Maybe we need to add a parameter which lets you do a deployment regardless of healthstate

**Pull request merged**